_inc/lib/core-api/wpcom-endpoints/service-api-keys.php

```diff
@@ -268,6 +284,83 @@ public static function validate_service_api_key_mapbox( $key ) {
 		);
 	}
 
+	/**
+	 * Get the site's own Mapbox API key if set, or the WordPress.com's one otherwise.
+	 *
+	 * @return array An array containing the key (if any) and its source ("site" or "wpcom").
+	 */
+	public static function get_service_api_key_mapbox() {
+		// If the site provides its own Mapbox API key, return it.
+		$service_api_key = Jetpack_Options::get_option( self::key_for_api_service( 'mapbox' ) );
+		if ( $service_api_key ) {
+			return self::format_api_key( $service_api_key );
+		}
+
+		// If the site is not WordPress.com, return an empty API key.
+		$site_id = self::get_wpcom_site_id();
+		if ( ( ! self::is_wpcom() && ! jetpack_is_atomic_site() ) || ! $site_id ) {
+			return self::format_api_key();
+		}
+
+		// If there is a cached token, return it.
+		$transient_key = 'wpcom_mapbox_access_token';
+		$cached_token  = get_transient( $transient_key );
+		if ( $cached_token ) {
+			return self::format_api_key( $cached_token, 'wpcom' );
+		}
+
+		// Otherwise retrieve a WordPress.com token.
+		$request_url = 'https://public-api.wordpress.com/wpcom/v2/sites/' . $site_id . '/mapbox';
+		$response    = wp_remote_get( esc_url_raw( $request_url ) );
```

Comment:
As this exists, anyone who knows the site id of any wpcom site can use this endpoint to retrieve the API key--it doesn't have to be their site. I wonder if this is a good use case for `wpcom_json_api_request_as_blog` to authenticate the request for that endpoint?

Additionally, the key could be accessed directly if `IS_WPCOM`, saving the overhead of the API request.

Reply:
Well, we kinda have to assume that the WPCOM token is publicly accessible anyway.
There are countless of ways to obtain it, even without an API request. Basically you just need a free WPCOM site and the browser dev tools. 😅 

I've chosen to have soft checks for this reason, and also because the token is used on the front end as well, possibly requested by unauthenticated visitors.